src/translate.c

```diff
@@ -213,19 +213,18 @@ static SEXP chr_translate_encoding(SEXP x, R_len_t size) {
   const SEXP* p_x = STRING_PTR_RO(x);
 
   SEXP out = PROTECT(r_clone_referenced(x));
-  SEXP* p_out = STRING_PTR(out);
 
   const void *vmax = vmaxget();
 
   for (int i = 0; i < size; ++i) {
     SEXP chr = p_x[i];
 
     if (Rf_getCharCE(chr) == CE_UTF8) {
-      p_out[i] = chr;
+      SET_STRING_ELT(out, i, chr);
```

Comment:
This is the most common case when translating (do nothing), and `out` is just a clone of `x`, so I imagine that we could probably just skip this entirely (for performance). Not saying you have to make that change in this PR, just something I noticed that we might want to revisit.